panoptes_client/workflow.py

```diff
@@ -179,6 +182,343 @@ def subject_workflow_statuses(self, subject_set_id):
+        self.add_extractor('external', 'alice', alice_task_key, alice_extractor_attributes)
+
+    def add_alice_reducers(self, alice_min_views=5, low_consensus_threshold=3):
+        """
+        Adds ALICE Reducers for given workflow.
+        (3 reducers created; an External Reducer, a StatsReducer, and a CountReducer)
+        -**alice_min_views** -This is the threshold number of classifications in order to "gray-out" a transcribed line.
+        Default is 5.
+        - **low_consensus_threshold** - This is the threshold number of classifications in agreement for good consensus.
+        Default is 3
+        """
+        external_reducer_url = 'https://aggregation-caesar.zooniverse.org/reducers/optics_line_text_reducer'
+        if alice_min_views or low_consensus_threshold:
+            external_reducer_url += f'?minimum_views={alice_min_views}&'
+            external_reducer_url += f'low_consensus_threshold={low_consensus_threshold}'
+
+        default_filter_attribs = {
+            'repeated_classifications': 'keep_first'
+        }
+        external_reducer_attributes = {
+            'url': external_reducer_url,
+            'filters': {
+                'extractor_keys': ['alice'],
+                **default_filter_attribs
+            }
+        }
+        self.add_reducer('external', 'alice', external_reducer_attributes)
+
+        complete_reducer_attribs = {
+            'filters': {
+                'extractor_keys': ['complete'],
+                **default_filter_attribs
+            }
+        }
+        self.add_reducer('stats', 'complete', complete_reducer_attribs)
+
+        self.add_reducer('count', 'count', complete_reducer_attribs)
+
+    def add_alice_rules_and_effects(self, question_retirement_limit=3, count_retirement_limit=30):
+        """
+        Adds Subject rules and its corresponding effects for ALICE configuration for the given workflow.
+        (2 subject rules will be created that will trigger retirement [Question rule and a Count rule])
+        - A total of 4 subject rule effects should get created.
+        -  There should be 2 effects related to the Question Rule condition
+        (one to send to ALICE and the othher to retire subject)
+        - There should also be 2 effects related to the Count Rule condition
+        (one to send to alice and the other to retire subject)
+        -**question_retirement_limit** Question subject rule created will trigger retirement when the answer to:
+        "is this complete" question reaches this threshhold limit (defaults to 3)
+        - **count_retirement_limit**
+        Count Subject Rule created will trigger retirement when the classification count reaches this limit
+
+        """
+        question_subject_rule = self.add_rule(
+            f'["gte", ["lookup", "complete.0", 0], ["const", {question_retirement_limit}]]',
+            'subject'
+        )
+        send_to_alice_effect_config = {
+            'url': 'https://tove.zooniverse.org/import',
```

Comment:
Noting that hard coding this here prevents using this functionality against the staging server, but that's not blocking.

Reply:
True, technically one cannot do specific ALICE reducer/extractor/rules/effects to caesar-staging via `workflow.py`'s caesar methods. I was debating on whether or not to add another parameter within workflow to add caesar connection or caesar endpoint. I decided against it.

A workaround is to connect via `caesar.py`  and to create reducers/extractors/rules/effects by hand. 
i.e. 
```
caesar = Caesar(endpoint='https://caesar-staging.zooniverse.org')
# This will add workflow to caesar staging 
caesar.add_workflow(_YOUR_WORKFLOW_ID, True, True)

```

Reply:
Cool, glad there's a workaround for testing, that works for me.